File: Backend/Models/Post.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models;

public class Post
{
    public int Id { get; set; }

    [Required, MaxLength(180)]
    public string Title { get; set; } = default!;

    [Required, MaxLength(200)]
    public string Slug { get; set; } = default!;

    [Required]
    public string Content { get; set; } = default!;

    [Required]
    public string AuthorId { get; set; } = default!;

    [ForeignKey(nameof(AuthorId))]
    public AppUser? Author { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    public bool IsPublished { get; set; } = true;
}

File: Frontend/src/type/post.ts
export type PostItem = {
    id: number;
    title: string;
    slug: string;
    content: string;
    isPublished: boolean;
    createdAt: string;
    updatedAt?: string | null;
    authorId: string;
    authorEmail?:string | null;
    authorFullName?:string | null;
    authorAvatarUrl?:string | null;
}

export type PostCreateRequest = {
    title: string;
    content: string;
    isPublished: boolean;
}

export type PostUpdateRequest = PostCreateRequest;